### PR TITLE
Sort the content of `dev` and `etc` directories

### PR DESCRIPTION
**Pull request by fingolfin**
The idea is that `etc` is shipped in GAP releases and contains extras
for users, while `dev` is not shipped and only for people interacting
with the GAP git repository. Hence move CI specific helper scripts as
well as `bisect.sh` from `etc` to `dev`.

Conversely, move some ctags related files from `dev` to `etc`, as those are
needed to let `make tags` function correctly.

Move `download.sh` from `etc` to `cnf` as it is used by the build system,
not users.

Finally, add `etc/README.md` explaining some of the files there.

(This doesn't touch or explain `etc/travis_fastfail.sh` as that is removed in #4263 which I made before realizing there should be bigger changes).

### CODE REVIEW

**Review by alex-konovalov — APPROVED** [review]
Very good! (I wonder if we need to put them in `dev/scripts` or something like that).

BTW, there is still `dev/rsyncserv` directory there - @frankluebeck do you think we need it there?

**Review by wilfwilson — APPROVED** [review]

**Comment by frankluebeck** [comment]
@alex-konovalov : The directory  `dev/rsyncserv` is outdated and can now be deleted.
My new attempt to revive the rsync distribution looks good. If it works I will create a new git-repository for the code.

File: etc/README.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+## Files for GAP developers
+
+The following are tools for people working on the GAP code base itself
+
+- `ffgen.c`: used by the GAP build system
+- `tags.sh`: used by the `make tags` build target
+
+## Files for GAP package authors
+
+- `Makefile.gappkg`
```
Comment (ssiccha) [inline etc/README.md:15]: What is this file for? That's missing.

Comment (fingolfin) [reply etc/README.md]: Added a description

File: etc/README.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+The following are tools for people working on the GAP code base itself
+
+- `ffgen.c`: used by the GAP build system
+- `tags.sh`: used by the `make tags` build target
+
+## Files for GAP package authors
+
+- `Makefile.gappkg`: build rules intended for use by GAP packages with a kernel extension
+- `convert.pl`: script to convert old-style GAP manual TeX files to HTML
```
Comment (alex-konovalov) [inline etc/README.md:16]: Perhaps emphasise and say "the old style gapmacro-based GAP manual TeX files"?

**Comment by alex-konovalov** [comment]
@fingolfin please backport this - I can then update gap-distribution release wrapping scripts (easy fix there, and when new release tarballs will appear, we can compare results of two release processes to test).

**Comment by fingolfin** [comment]
Backported to stable-4.11 in 94dfe32aa0dc2403b54b3c74131b02129e7c9f42

**Pull request merged**